ios/brave-ios/Sources/Brave/Frontend/Settings/General/SearchEngines/SearchSettingsTableViewController.swift

```diff
@@ -469,6 +478,13 @@ class SearchSettingsTableViewController: UITableViewController {
   override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
     return indexPath.section == Section.customSearch.rawValue
   }
+
+  override func setEditing(_ editing: Bool, animated: Bool) {
+    super.setEditing(editing, animated: animated)
+
+    // Hide the left bar button until edit action is finished
+    navigationItem.leftBarButtonItem = isPresentedModally && !editing ? dismissBarButton : nil
```

Comment:
Call `setLeftBarButtonItem(_:animated:)` instead so it animates nicely

Reply:
fixed at https://github.com/brave/brave-core/pull/24413/commits/c10b763d1049c3a2ba024da8321eec1943a20229